lib/GenomeFileUtil/core/GenbankToGenome.py

```diff
@@ -516,15 +515,7 @@ def _save_assemblies(self, workspace_id, genome_objs):
             contigs = Bio.SeqIO.parse(genome_obj.consolidated_file, "genbank")
             genome_obj.assembly_id = f"{genome_obj.genome_name}_assembly"
             genome_obj.assembly_path = f"{self.cfg.sharedFolder}/{genome_obj.assembly_id}.fasta"
```

Comment:
It seems like these lines are occurring twice for each genome. Is that expected?

Reply:
2 cases, cannot be both

1. when an assembly ref is provided
2. when an assembly ref is not provided.

Reply:
It's now always called in `_load_contigs` and also called if there's no assembly ref

Reply:
so which line is causing confusion?

Reply:
Why we're parsing the file twice

Reply:
1. load contigs
2. create FASTA files for bulk upload (https://github.com/kbaseapps/GenomeFileUtil/blob/1fc6ed1165094bfc528efc52f9415ae9519c39d8/lib/GenomeFileUtil/core/GenbankToGenome.py#L522)

Reply:
Why not create the files when you load the contigs?

Reply:
The trade-off is between Bio.SeqIO.parse and Bio.SeqIO.write. Creating the file while loading the contigs avoids parsing in _save_assemblies, but it may also result in the creation of an unused file. After considering the options, I decided to create the file only when necessary.

This was my thought process.

Reply:
How much time do the parses and writes take, approximately? And how large of a file?

Reply:
<img width="1515" alt="Screenshot 2024-08-13 at 11 58 18 AM" src="https://github.com/user-attachments/assets/61f5c71a-fefc-4825-80fc-43d17d5b496d">

Reply:
`GCF_000970165.1_ASM97016v1_genomic.gbff` is from NCBI.

Reply:
Ok, so writing >> reading. So reading twice is not going to slow things down much compared to the write. Can you add a quick note to that effect at the 2nd read?

Reply:
👍

Reply:
:+1: